warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java

```diff
@@ -279,14 +284,79 @@ protected ASTJexlScript pushdownLargeFieldedLists(ShardQueryConfiguration config
         if (config.canHandleExceededValueThreshold()) {
             URI hdfsQueryCacheUri = getFstHdfsQueryCacheUri(config, settings);
             
+            Map<String,Integer> pushdownCapacity = new HashMap<>();
+            ASTJexlScript script;
             if (hdfsQueryCacheUri != null) {
                 FileSystem fs = VisitorFunction.fileSystemCache.getFileSystem(hdfsQueryCacheUri);
-                // Find large lists of values against the same field and push down into
-                // an Ivarator
-                return PushdownLargeFieldedListsVisitor.pushdown(config, queryTree, fs, hdfsQueryCacheUri.toString());
+                // Find large lists of values against the same field and push down into an Ivarator
+                script = PushdownLargeFieldedListsVisitor.pushdown(config, queryTree, fs, hdfsQueryCacheUri.toString(), pushdownCapacity);
             } else {
-                return PushdownLargeFieldedListsVisitor.pushdown(config, queryTree, null, null);
+                script = PushdownLargeFieldedListsVisitor.pushdown(config, queryTree, null, null, pushdownCapacity);
             }
+            
+            // check term limits and use the capacity map to reduce further if necessary
+            int termCount = TermCountingVisitor.countTerms(script);
+            if (termCount > config.getMaxTermThreshold()) {
+                // check if the capacity is available to get under the term limit
+                // determine if its possible to reduce enough to meet the threshold
+                int capacitySum = 0;
+                for (Integer capacity : pushdownCapacity.values()) {
+                    capacitySum += capacity;
+                }
+                
+                if (termCount - capacitySum <= config.getMaxTermThreshold()) {
+                    // preserve the original config and set minimum thresholds for collapsing Ors, do not use FST for performance reasons
```

Comment:
This comment strikes me as odd.  Technically we aren't doing anything programmatically to prevent FSTs from running.  If the FST threshold is set to 2, you will absolutely create an FST instead of a list ivarator.

Reply:
yep, thanks